### PR TITLE
Fixes for category filter

### PR DESCRIPTION
**Pull request by charypar**
Addresses #224.

## Tasks
* [x] make entire category filter header reveal the filter screen
* [x] make category pills on event detail pages navigate to a pre-filtered events screen

The only worry I have with this is that the navigation from the detail page is "destructive" - there is no way for the user to go back to the event they were just looking at. Maybe something to address later on, because it might make us re-think the navigation quite a bit.

## Pre-flight check-list

Before raising a pull request

* [ ] Documentation
* [x] Unit tests

## Pre-merge check-list

* [x] Link to Trello ticket/GitHub issue (If applicable)
* [ ] Any risky areas identified
* [ ] Test plan provided
* [ ] Tester approved

## Tester check-list

* [ ] Tested on multiple devices / OS versions (Test on the physical device(s) you have access to, otherwise use BrowserStack):

  * [ ] Galaxy S8 (Android 7.0/8.0)
  * [ ] Galaxy S7 (Android 6.0)
  * [ ] iPhone X (iOS 11.x)
  * [ ] iPhone 8 (11.x)
  * [ ] iPhone 7 (iOS 10.x)

  Optional:

  * [ ] Google Pixel 2
  * [ ] Galaxy S8 Plus
  * [ ] Galaxy S7 Edge
  * [ ] Galaxy S6
  * [ ] iPhone 7+
  * [ ] iPhone 6

Accessibility Testing (If applicable)

* [ ] Text-to-Voice (Android: Voice Assistant / iOS: Speak Selection)
* [ ] Large Text (=<200%)
* [ ] Landscape Screen orientation

Weak connection testing (If applicable)

* [ ] Airplane mode
* [ ] 2G/3G Network Settings

### CODE REVIEW

File: src/components/FilterHeaderCategories.js [hunk 1]
```diff
@@ -53,18 +54,19 @@ const FilterHeaderCategories = ({ selectedCategories, onFilterPress }: Props) =>
           {text.filterTitle}
         </Text>
       )}
-      <View style={styles.interestButton}>
+      <Touchable style={styles.interestButton} onPress={onFilterPress}>
         <Text type="h2" style={styles.interestButtonText}>
           {text.filterByInterest}
         </Text>
         <CategoriesFilterButton onPress={onFilterPress} />
```
Comment (frigus02) [inline src/components/FilterHeaderCategories.js]: Is the `onPress` here still needed?

File: src/components/NavigationTabBar.js [hunk 2]
```diff
@@ -33,19 +32,7 @@ class NavigationTabBar extends React.PureComponent<_TabBarBottomProps> {
...
-      const childRoute = navigation.state.routes[index];
-      if (typeof childRoute.index === "number" && childRoute.index > 0) {
-        navigation.dispatch(
-          NavigationActions.popToTop({ key: childRoute.key })
-        );
-      }
-    } else {
-      jumpToIndex(index);
-    }
+    this.props.jumpToIndex(index);
```
Comment (frigus02) [inline src/components/NavigationTabBar.js]: Hm nice, I guess we won't need this behaviour anymore. All subviews hide the tab bar anyway.

File: src/components/FilterHeaderCategories.js (hunk 1 above)
Comment (charypar) [reply src/components/FilterHeaderCategories.js]: It still provides a nice feedback, if you tap it directly.

File: src/components/NavigationTabBar.js (hunk 2 above)
Comment (charypar) [reply src/components/NavigationTabBar.js]: Yea, that was my thinking

**Pull request merged**